tests/publiccloud/az_accelerated_net.pm

```diff
@@ -40,9 +42,13 @@ a VM again, since the IPs will differ.
 
 =cut
 sub get_new_ip {
-    my ($self, $instance) = @_;
-    assert_script_run('az vm list -g ' . $instance->instance_id . ' -d');
-    my $cmd = 'az vm list -g ' . $instance->instance_id . q( -d|grep -i publicip|awk '{print $2}'| tr -d '"'| tr -d ',');
+    my ($instance, $need_pub) = @_;
+    my $resgroup = (split('/', $instance->instance_id))[4];
+    assert_script_run("az vm list -g $resgroup -d");
+    my $cmd = 'az vm list-ip-addresses --ids ' . $instance->instance_id . ' --query "[].virtualMachine.network.privateIpAddresses[]" -o tsv';
+    if (!!$need_pub) {
+        $cmd = 'az vm list-ip-addresses --ids ' . $instance->instance_id . ' --query "[].virtualMachine.network.publicIpAddresses[].ipAddress" -o tsv';
```

Comment:
Firstly, the test deallocates and allocates the VMs, not sure whether this might not change the public IPs in general. At least I have noticed that the `$instance->public_ip` was not referring to a working IP and ssh connections died.

Reply:
ok , make sense .